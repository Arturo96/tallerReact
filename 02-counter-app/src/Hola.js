import React from "react";
import PropTypes from "prop-types";

const Hola = ({ saludo, subtitle }) => {
	return (
		<>
			<h1>{saludo}</h1>
			<p>{subtitle}</p>
		</>
	);
};

Hola.propTypes = {
	saludo: PropTypes.string.isRequired
};

Hola.defaultProps = {
	subtitle: "Random"
};

export default Hola;
