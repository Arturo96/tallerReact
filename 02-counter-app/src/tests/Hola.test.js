import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import React from 'react';
import Hola from '../Hola';

describe('Pruebas del componente Hola', () => {
    // test('debe de mostrar el mensaje Hola Arturo', () => {
    //     const saludo = "Hola Arturo";
    //     const { getByText} = render(<Hola saludo={saludo} />);
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('debe de mostrar <Hola /> correctamente ', () => {

        const saludo = 'Hola Arturo';
        
        const wrapper = shallow(<Hola saludo={saludo}/>);

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola Arturo';
        const subTitle = 'Esto es un subtitulo';
        
        const wrapper = shallow(<Hola 
            saludo={saludo}
            subtitle={subTitle}
            />);

        const elParrafo = wrapper.find('p').text();
        expect(elParrafo).toBe(subTitle);
    })
    
    
    
})