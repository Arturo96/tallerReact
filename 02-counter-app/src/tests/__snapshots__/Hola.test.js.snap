// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente Hola debe de mostrar <Hola /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola Arturo
  </h1>
  <p>
    Random
  </p>
</Fragment>
`;
